compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -58,6 +58,7 @@ class CompilationTests extends ParallelTesting {
     compileFile("tests/pos-special/i3323.scala", defaultOptions.and("-Xfatal-warnings")) +
     compileFile("tests/pos-special/i3323b.scala", defaultOptions.and("-Xfatal-warnings")) +
     compileFile("tests/pos-special/i3589-b.scala", defaultOptions.and("-Xfatal-warnings")) +
+    compileFile("tests/pos-special/i4006.scala", defaultOptions.and("-Ytest-pickler")) +
```

Comment:
Why not put the test in `tests/pickling`?

Reply:
Yes